pkg/cloudprovider/metrics/cloudprovider.go

```diff
@@ -71,30 +91,77 @@ func Decorate(cloudProvider cloudprovider.CloudProvider) cloudprovider.CloudProv
 
 func (d *decorator) Create(ctx context.Context, machine *v1alpha5.Machine) (*v1alpha5.Machine, error) {
 	defer metrics.Measure(methodDurationHistogramVec.WithLabelValues(injection.GetControllerName(ctx), "Create", d.Name()))()
-	return d.CloudProvider.Create(ctx, machine)
+	machine, err := d.CloudProvider.Create(ctx, machine)
+	if err != nil {
+		errorsTotalCounter.With(prometheus.Labels{
```

Comment:
I think we need the controller name label defined throughout all the `errorTotalCounter` metrics here

Reply:
I didn't fully understand what this comment means so I'm not quite sure how to handle that label

```
// The value used for the metric label, "controller", is taken from the `Context` object
// passed to the methods of `CloudProvider`.
```

is the val a simple dereference of something in `ctx`?

Reply:
nevermind, I think I figured out that it must be `injection.GetControllerName(ctx),` from our `github.com/aws/karpenter-core/pkg/operator/injection` package